pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1309,3 +1340,111 @@ func initializeAnnotations(cd *hivev1.ClusterDeployment) {
 		cd.Annotations = map[string]string{}
 	}
 }
+
+// mergePullSecrets takes pull secretes in to account and returns the merged pull request
```

Comment:
Suggest: // mergePullSecrets merges the global pull secret JSON (if defined) with the cluster's pull secret JSON (if defined). An error will be returned if neither is defined.